geomstats/_backend/tensorflow/autodiff.py

```diff
@@ -163,3 +163,49 @@ def jac(x):
         return g.jacobian(y, x)
 
     return jac
+
+
+def hessian(func):
+    """Return a function that returns the hessian of func.
+
+    Parameters
+    ----------
+    func : callable
+        Function whose Hessian is computed.
+
+    Returns
+    -------
+    hess : callable
+        Function taking x as input and returning
+        the hessian of func at x.
+    """
+
+    def hess(x):
+        """Return the hessian of func at x.
+
+        Parameters
+        ----------
+        x : array-like
+            Input to function func or its hessian.
+
+        Returns
+        -------
+        _ : array-like
+            Value of the hessian of func at x.
+        """
+        # Note: this is a temporary implementation
+        # that uses the jacobian of the gradient.
+        # inspired from https://github.com/tensorflow/tensorflow/issues/29781
+        # waiting for the hessian function to be implemented in GradientTape.
+        if isinstance(x, _np.ndarray):
+            x = _tf.Variable(x)
```

Comment:
There is some optimization that is done with scipy, and for which we convert back to numpy... :/
Not the best, I agree. Probably not something to tackle in this PR though?

Reply:
OK, let's leave it for later. (btw, I've developed a solution for this `scipy.minimize` conversion problem. It is in the (new) PR for the solvers (which I haven't opened yet) - since some of them are basically optimization problems.)

Reply:
Awesome!!! :D :D